app/experimenter/experiments/serializers/design.py

```diff
@@ -33,7 +34,23 @@ def update_changelog(self, instance, validated_data):
 
 class PrefValidationMixin(object):
 
-    def validate_pref(self, pref_type, pref_value, field_name):
+    def validate_pref_branch(self, pref_branch):
+        if pref_branch == "Firefox Pref Branch":
+            return {"pref_branch": "Please select a branch"}
+        return {}
+
+    def validate_multi_preference(self, pref):
+        pref_type = pref["pref_type"]
+        pref_value = pref["pref_value"]
+        field_name = "pref_value"
+        errors = {}
+        if pref_type == "Firefox Pref Type":
+            errors["pref_type"] = "Please select a pref type"
+
+        errors.update(self.validate_pref_value(pref_type, pref_value, field_name))
+        return errors
```

Comment:
**Opinion:**

Just a super personal annoying style thing, for this type of pattern I usually do something like

        def something():
            thingtoReturn = {}  # initialize default return at the top of the function
            # buncha logic
            if whatever:
                thingToReturn[whatever] = whatever
            # buncha logic
            return thingToReturn  # last line returns the variable defined at the top


This makes it easier to go back and move things around without modifying that outer wrapping of setting up the default case and always returning it

Reply:
yup yup, I agree. I originally wanted to do this, but realized that the singular pref also relies on this and thought it would've been messy to change it. Turns out it wasn't so bad at all 🤷‍♀